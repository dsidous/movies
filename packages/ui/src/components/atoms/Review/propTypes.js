import { object } from 'prop-types';

export const propTypes = {
  review: object,
};

export const defaultProps = {};
