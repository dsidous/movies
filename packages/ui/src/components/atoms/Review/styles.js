import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      background: '#eee',
      borderRadius: 4,
      boxShadow: '0 0 40px rgba(0,0,0,0.1)',
      marginBottom: theme.spacing(5),
      padding: '35px 15px',
      width: '100%',
    },

    author: {
      fontSize: '4vw',
      fontWeight: 'bold',
      margin: '0 0 20px',
      [theme.breakpoints.up('sm')]: {
        fontSize: '16px',
      },
    },

    content: {
      display: 'flex',
      flexDirection: 'column',
      fontSize: '3vw',
      whiteSpace: 'pre-line',
      [theme.breakpoints.up('sm')]: {
        fontSize: '14px',
        lineHeight: 1.6,
      },
    },
  }),
  { name: 'review' },
);
