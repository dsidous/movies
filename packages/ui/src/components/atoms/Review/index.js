import React from 'react';
import { compiler } from 'markdown-to-jsx';

import { propTypes } from './propTypes';

const Review = ({ review: { id, author, content } }) => (
  <div className="reviews-element" key={id}>
    <input type="checkbox" className="review-content__more" id={id} />
    <div className="review-author">{`A review by ${author}`}</div>
    <div
      className={['review-content', content.length > 500 ? 'long' : ''].join(
        ' ',
      )}
    >
      {compiler(content.slice(0, content.indexOf(' ', 500)), {
        forceInline: true,
      })}
      <span className="review-content__hidden">
        {compiler(content.slice(content.indexOf(' ', 500)), {
          forceInline: true,
        })}
      </span>
    </div>
    {content.length > 500 && (
      // eslint-disable-next-line jsx-a11y/label-has-associated-control
      <label htmlFor={id} className="review-content__more-label" />
    )}
  </div>
);

Review.propTypes = propTypes;

export default Review;
