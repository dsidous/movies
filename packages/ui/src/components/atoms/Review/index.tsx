import React from 'react';
import { Box, Typography, Paper } from '@material-ui/core';

import LongCopy from '../LongCopy';
import { propTypes } from './propTypes';
import useStyles from './styles';
import { Review as PropReview } from '../../../types/movie';

interface Props {
  review: PropReview;
}

const Review = ({ review: { author, content } }: Props) => {
  const classes = useStyles();

  return (
    <Paper className={classes.root}>
      <Typography
        className={classes.author}
      >{`A review by ${author}`}</Typography>
      <Box className={classes.content}>
        <LongCopy content={content} />
      </Box>
    </Paper>
  );
};

Review.propTypes = propTypes;

export default Review;
