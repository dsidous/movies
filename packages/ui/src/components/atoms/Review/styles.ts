import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      background: '#eee',
      marginBottom: theme.spacing(5),
      padding: theme.spacing(6, 4),
      width: '100%',
    },

    author: {
      fontWeight: 'bold',
      margin: theme.spacing(0, 0, 2),
    },

    content: {
      display: 'flex',
      flexDirection: 'column',
      whiteSpace: 'pre-line',
    },
  }),
  { name: 'review' },
);
