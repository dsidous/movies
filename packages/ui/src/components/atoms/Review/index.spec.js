import React from 'react';
import { shallow } from 'enzyme';

import Review from '.';

const mockProps = {
  review: {
    id: 1,
    author: 'Author',
    content: 'content',
  },
};

describe('Atoms/Review', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Review {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
