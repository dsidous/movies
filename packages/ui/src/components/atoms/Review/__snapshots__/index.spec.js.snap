// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Review should render as expected 1`] = `ShallowWrapper {}`;
